### PR TITLE
destination-postgres: bump CDK

### PR DESCRIPTION
**Pull request by stephane-airbyte**
Refactored `PostgresDestination`, `PostgresSqlOperations`, and `PostgresDestinationHandler` to support the new `PostgresGenerationHandler` class for handling generation IDs.

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43331?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#43315** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43315?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42977** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42977?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43330** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43330?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43331** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#43329** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43329?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-postgres/metadata.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ data:
   connectorSubtype: database
   connectorType: destination
   definitionId: 25c5221d-dce2-4163-ade9-739ef790f503
-  dockerImageTag: 2.3.0
+  dockerImageTag: 2.3.1
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-postgres/metadata.yaml:8]: you need to rebump this, 2.3.1 got published earlier today 😅 https://github.com/airbytehq/airbyte/pull/43363

File: airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/PostgresGenerationHandler.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.postgres
+
+import io.airbyte.cdk.db.jdbc.JdbcDatabase
+import io.airbyte.cdk.integrations.destination.jdbc.JdbcGenerationHandler
+
+class PostgresGenerationHandler : JdbcGenerationHandler {
+    override fun getGenerationIdInTable(
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/PostgresGenerationHandler.kt:11]: didn't read this, I assume it's just moving code from pg sql ops to a new class

**Review by edgao — APPROVED** [review]
seems right, pending my comment in https://github.com/airbytehq/airbyte/pull/43329

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Aug 7, 5:49 PM EDT**: @stephane-airbyte started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/43331).

**Pull request merged**